src/libponyc/pass/refer.c

```diff
@@ -714,7 +721,43 @@ bool refer_qualify(pass_opt_t* opt, ast_t* ast)
   return true;
 }
 
-static bool assign_multi_dot(pass_opt_t* opt, ast_t* ast, bool need_value)
+static void error_check_used_decl(errorframe_t* frame, ast_t* ast)
+{
+  // Prints an info about why the lvalue is needed
+  ast_t* parent = ast_parent(ast);
+  pony_assert(parent != NULL);
+  token_id parent_id = ast_id(parent);
+
+  if (parent_id == TK_VAR || parent_id == TK_LET) {
+    ast_error_frame(frame, parent, "the previous value of '%s' is used because you are trying to use the return value of this %s declaration", ast_print_type(ast), ast_print_type(parent));
```

Comment:
```suggestion
    ast_error_frame(frame, parent, "the previous value of '%s' is used because you are trying to use the resulting value of this %s declaration", ast_print_type(ast), ast_print_type(parent));
```

Reply:
One small pedantic suggestion on the error message. Strictly speaking, "return value" would refer to the return value of a function/method, so "resulting value" is more generic and less confusing.